### PR TITLE
feat: Calendar support cssVar

### PR DESCRIPTION
**Pull request by RedJue**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- #45618 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Calendar support cssVar       |
| 🇨🇳 Chinese |    日历组件支持 css 变量       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 73b4070</samp>

This pull request refactors the calendar component to use CSS variables for theming, instead of static styles. It introduces a new hook `useCSSVar` that handles the registration and update of the CSS variables based on the theme token. It also updates the component's style file to use the theme token system and the `@ant-design/cssinjs` library.

### <samp>🤖 Generated by Copilot at 73b4070</samp>

*  Generate and use a custom hook for registering and updating CSS variables for the calendar component based on the theme token ([link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-841e97dfb23daba72a34ee15810c8b36da6464eba16e120739ec6c9636f94fb2R19), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-841e97dfb23daba72a34ee15810c8b36da6464eba16e120739ec6c9636f94fb2L122-R124), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-841e97dfb23daba72a34ee15810c8b36da6464eba16e120739ec6c9636f94fb2L284-R286), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-2cc1bcdf491496d5b42c2b5be2158d39e18ba2bb626260237284a87be965a8aaR1-R4), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544R218-R226))
  * Import the `useCSSVar` hook from `./style/cssVar` in `components/calendar/generateCalendar.tsx` ([link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-841e97dfb23daba72a34ee15810c8b36da6464eba16e120739ec6c9636f94fb2R19))
  * Destructure the `hashId` from the `useStyle` hook and call the `useCSSVar` hook with the `prefixCls` as the argument in `components/calendar/generateCalendar.tsx` ([link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-841e97dfb23daba72a34ee15810c8b36da6464eba16e120739ec6c9636f94fb2L122-R124))
  * Replace the `wrapSSR` function with the `wrapCSSVar` function returned by the `useCSSVar` hook in `components/calendar/generateCalendar.tsx` ([link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-841e97dfb23daba72a34ee15810c8b36da6464eba16e120739ec6c9636f94fb2L284-R286))
  * Create a new file `components/calendar/style/cssVar.ts` that exports the `useCSSVar` hook as the default export ([link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-2cc1bcdf491496d5b42c2b5be2158d39e18ba2bb626260237284a87be965a8aaR1-R4))
  * Generate the `useCSSVar` hook by calling the `genCSSVarRegister` function from `../../theme/internal` with the component name and the `prepareComponentToken` function as the arguments in `components/calendar/style/cssVar.ts` ([link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-2cc1bcdf491496d5b42c2b5be2158d39e18ba2bb626260237284a87be965a8aaR1-R4), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544R218-R226))
  * Define the `prepareComponentToken` function that takes a theme token and returns a default token for the calendar component in `components/calendar/style/index.ts` ([link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544R218-R226))
* Convert the numeric token values to CSS unit strings with the `unit` function from `@ant-design/cssinjs` and replace the hard-coded numeric values with the `token.calc` function in `components/calendar/style/index.ts` ([link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544R2-R3), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L9-R11), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L65-R67), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L82-R84), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L88-R90), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L101-R103), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L108-R110), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L130-R132), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L164-R173), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L191-R193), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L199-R201), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L228-R245), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L236-R251))
  * Import the `unit` function and the `GetDefaultToken` type from `@ant-design/cssinjs` and `../../theme/internal` respectively in `components/calendar/style/index.ts` ([link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544R2-R3), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L9-R11))
  * Wrap the `token.paddingSM`, `token.lineWidth`, `token.paddingXS`, `token.borderRadiusLG`, `token.weekHeight`, `token.marginXS`, `token.lineWidthBold`, `token.dateValueHeight`, and `token.screenXS` values with the `unit` function in `components/calendar/style/index.ts` ([link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L65-R67), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L82-R84), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L88-R90), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L101-R103), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L108-R110), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L130-R132), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L164-R173), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L191-R193), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L199-R201))
  * Replace the hard-coded numeric values for the `weekHeight` and `dateContentHeight` properties with the `token.calc` function in `components/calendar/style/index.ts` ([link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L228-R245), [link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L236-R251))
  * Replace the inline function that returns the default token for the calendar component with the `prepareComponentToken` function in `components/calendar/style/index.ts` ([link](https://github.com/ant-design/ant-design/pull/45835/files?diff=unified&w=0#diff-301b86a4e3e8d0c690868d2f70adcef037741d8db81cdad1dcc989a4c8be0544L236-R251))

### CODE REVIEW

File: components/calendar/style/index.ts [hunk 1]
```diff
@@ -225,20 +236,17 @@ export default genComponentStyleHook(
         calendarCls,
         pickerCellInnerCls: `${token.componentCls}-cell-inner`,
         dateValueHeight: token.controlHeightSM,
-        weekHeight: token.controlHeightSM * 0.75,
-        dateContentHeight:
-          (token.fontSizeSM * token.lineHeightSM + token.marginXS) * 3 + token.lineWidth * 2,
+        weekHeight: token.calc(token.controlHeightSM).mul(0.75).equal() as number,
+        dateContentHeight: token
+          .calc(token.calc(token.fontHeightSM).add(token.marginXS).equal())
```
Comment (MadCcc) [inline components/calendar/style/index.ts:241]:
```suggestion
          .calc(token.calc(token.fontHeightSM).add(token.marginXS))
```
最后再 equal 就行

File: components/calendar/style/index.ts [hunk 2]
```diff
@@ -225,20 +236,17 @@ export default genComponentStyleHook(
...
+          .mul(3)
+          .add(token.lineWidth)
+          .mul(2)
```
Comment (MadCcc) [inline components/calendar/style/index.ts:244]: 这里要括起来，链式调用不会按照符号优先级

**Comment by MadCcc** [comment]
argos 里对应组件需要保证没有变化

**Comment by RedJue** [comment]
ok

File: components/calendar/style/index.ts (hunk 1 above)
Comment (RedJue) [reply components/calendar/style/index.ts:241]: 1

File: components/calendar/style/index.ts (hunk 2 above)
Comment (RedJue) [reply components/calendar/style/index.ts:244]: ok

File: components/calendar/generateCalendar.tsx [hunk 3]
```diff
@@ -119,7 +120,8 @@ function generateCalendar<DateType>(generateConfig: GenerateConfig<DateType>) {
     const prefixCls = getPrefixCls('picker', customizePrefixCls);
     const calendarPrefixCls = `${prefixCls}-calendar`;
 
-    const [wrapSSR, hashId] = useStyle(prefixCls);
+    const [, hashId] = useStyle(prefixCls);
+    const wrapCSSVar = useCSSVar(prefixCls);
```
Comment (MadCcc) [inline components/calendar/generateCalendar.tsx:124]:
```suggestion
    const wrapCSSVar = useCSSVar(calendarPrefixCls);
```
prefixCls 没有实际用过

**Review by MadCcc — APPROVED** [review]

**Pull request merged**